www/data_module/src/services/indexeddb/indexeddb.service.coffee

```diff
@@ -0,0 +1,208 @@
+class Ronron extends Run
+    constructor: (dataService) ->
+        # dataService.get('builders/1/builds/1', {builderid: 1}).then (a) -> console.log a
+
+class DBStores extends Constant
+    constructor: ->
+        return {
+            paths: '&[path+query],path,query,tracked,lastActive'
+        }
+
+class IndexedDB extends Service
+    constructor: ($log, $window, $injector, $q, dataUtilsService, DBSTORES, SPECIFICATION) ->
+        return new class IndexedDBService
+            constructor: ->
+                @db = new $window.Dexie('BBCache')
+                dbStores = angular.extend {},
+                    @processSpecification(SPECIFICATION), DBSTORES
+                @db.version(1).stores(dbStores)
+
+                # global db error handler
+                @db.on 'error', (e) -> $log.error(e)
+
+                @db.open()
+
+            get: (url, query = {}) ->
+                [tableName, q, id] = @processUrl(url)
+                angular.extend query, q
+
+                table = @db[tableName]
+
+                @db.transaction 'r', table, =>
+
+                    # convert promise to $q implementation
+                    if id? then return $q.resolve table.get(id)
+
+                    # TODO note: this takes almost a second
+                    promise = table.toArray()
+
+                    # 1. filtering
+                    filters = []
+                    for fieldAndOperator, value of query
+                        if ['field', 'limit', 'offset', 'order'].indexOf(fieldAndOperator) < 0
+                            filters[fieldAndOperator] = value
+                    promise = @filter(promise, filters)
+
+                    # 2. sorting
+                    order = query?.order
+                    promise = @sort(promise, order)
+
+                    # 3. pagination
+                    offset = query?.offset
+                    limit = query?.limit
+                    promise = @paginate(promise, offset, limit)
+
+                    # TODO 4. properties
+                    property = query?.property
+                    promise = @properties(promise, property)
+
+                    # 5. fields
+                    fields = query?.field
+                    promise = @fields(promise, fields)
+
+                    return promise
+
+            filter: (promise, filters) ->
```

Comment:
this does not look right to have a function which takes a promise, and do stuff on its results